kimchi/src/prover.rs

```diff
@@ -517,11 +382,193 @@ where
             // interpolating
             let aggreg8 = aggreg_coeffs.evaluate_over_domain_by_ref(index.cs.domain.d8);
 
-            lookup_context.aggreg_comm = Some(aggreg_comm);
-            lookup_context.aggreg_coeffs = Some(aggreg_coeffs);
-            lookup_context.aggreg8 = Some(aggreg8);
+            self.aggreg_comm = Some(aggreg_comm);
+            self.aggreg_coeffs = Some(aggreg_coeffs);
+            self.aggreg8 = Some(aggreg8);
```

Comment:
maybe it'd make more sense for this function to return a `Option<PolyData>`, where `PolyData` is a struct like

```
PolyData {
  commitment: PolyComm<_>,
  coefficient: DensePolynomial<_>,
  evaluation_d8: Evaluations<_>,
}
```

and we can remove these fields from the context (ideally the context would disappear).

But then again, I'm not sure it makes sense to not keep that logic inlined within the prover.

Reply:
Agree, but this change is meant to be very specific (see by big reply above) and not a proper refactor.